refactor: Migrate to AndroidX

haroldadmin opened the pull request:
### Description

This PR migrates all the dependencies in the project to the AndroidX namespace for future maintainability.

Additionally:
- Updated Gradle version.
- Fixed recycler view's constraint layout settings

Fixes #160 

### Type of Change:
- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Ran the app on personal device.

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### Notes
With the migration to AndroidX, explicit dependencies on v4 and v7 support libraries are redundant, and cause compilation errors so I have removed them. I have also added additional `jcenter()` repository in the buildscript for `buildSrc` module to fix compilation errors. More information can be found [here](https://github.com/gradle/kotlin-dsl/issues/1033)

isabelcosta commented:
@haroldadmin I merged some PRs that caused you PR to have merge conflicts. Please fix that, and then I can review your PR

haroldadmin commented:
@isabelcosta I have fixed the merge conflict and the build errors. This is a massive PR, but most of the changes are in package names only. Please review when you can 😊

isabelcosta commented on buildSrc/build.gradle.kts:
```diff
@@ -1,3 +1,12 @@
+ */
+repositories {
+    jcenter()
+}
```
fix this no line at the end of file problem

haroldadmin replied:
Whoops. Thanks for pointing this out.

haroldadmin replied:
Fixed now.

isabelcosta commented on build.gradle:
```diff
@@ -1,7 +1,7 @@
 
 buildscript {
-
+    ext.kotlin_version = '1.3.21'
```
ist this kotlin version defined somewhere else on Dependencies file? Why are you explicitly putting it here?

haroldadmin replied:
I'm not sure why I did this. It's been some time since I made these changes. I'll investigate and let you know soon.

haroldadmin replied:
I used the automated option to update Kotlin, and it added this line here. I've fixed this now. Thanks for pointing this out!

isabelcosta commented on app/src/main/res/layout/fragment_mentorship_tasks.xml:
```diff
@@ -66,16 +66,18 @@
-        android:layout_marginEnd="16dp"
-        app:layout_constraintTop_toBottomOf="@id/tvAchievements"/>
+        android:layout_marginStart="8dp"
+        android:layout_marginEnd="8dp"
```
this is a change in the UI right? Does it need to be here? If you think it is, be explicit about this o nthe PR description, show the changes visually (through an image for example)

haroldadmin replied:
Actually I ran into a similar problem here as with my last PR. This particular recyclerview in question was not constrained properly, and this made the lint check fail. So I added these contraints myself. I'm not sure why these changes were merged without proper constraints in the first place.

isabelcosta replied:
Ok, I asked you to add this piece of change to the PR description :) I'll add it

isabelcosta reviewed (changes requested):
Thank you for your contribution here :)
Left some comments for you to address.

You did more than just adding support to AndroidX right? Be clear about that on the PR description. For example you updated gradle and other libraries versions, you also did a small UI change.

haroldadmin committed:
refactor: Migrate to AndroidX

This commit maps all existing dependency packages to their AndroidX
namespace equivalents. Redundant dependencies on v4 and v7 versions of
the support library have also been removed. BuildSrc module buildscript
has been modified to include additional required dependencies

haroldadmin commented:
@isabelcosta I have fixed the issues you pointed out. Please review when you can.

isabelcosta reviewed (approved):
Thank you @haroldadmin for your contribution!
